Draft-Accepted/RFCNNNN-Install-PowerShell.md

```diff
@@ -0,0 +1,218 @@
+- Updating of PowerShell (Update-PowerShell7)
+- Disconnect scenarios - Users that are air-gapped may download the appropriate release from GitHub.
+- Handling of additional MSI switch options - i.e. Enable PSRemoting
+
+## Specification
+
+Cmdlet name: Install-PowerShell7
+
+Alias: None
+
+The default behavior is to download and extract the .ZIP package, then install the latest stable
+release of PowerShell into the user scope allowed by permissions. If with administrative privilege,
+install to the scope of AllUsers. If user does not have administrative privilege, install to the
+scope CurrentUser.
+
+- The package install location - $env:LOCALAPPDATA\Microsoft\PowerShell
+- The installer should update the Users PATH for pwsh
+- The installer should enable Microsoft Updates for scope AllUsers. Due to permission requirements,
+  the scope CurrentUser will require manual updating.
+- Confirm impact should be set to **High** to force a warning and the ability to abort
+
+The following parameters can be added by experienced PowerShell users to customize the installation.
+
+- Parameter: Version - overrides the default and allows the user to specify the version of
+  PowerShell that will be installed. PowerShell version 7.0 and 7.1 are out of support, so version
+  numbers should begin with 7.2. This would install the latest 7.2.x.
+- Parameter: Scope - overrides the default and allows the user to specify the installation location
+  (scope) AllUsers or CurrentUser.
+- Parameter: Channel - options Stable/Preview/LTS - allows the user to select if they want the
+  current preview or Long Term Service (LTS) release. Preview should not be allowed if value of
+  parameter **-Version** is Generally Available (GA).
+- Parameter: Passthru - to provide informative output post installation
+
+## Syntax and Parameter Sets
+
+### ParameterSet Version (Default)
+
+```
+Install-PowerShell7 [[-Version] <string>] [-Scope {AllUsers | CurrentUsers}] [-Passthru] [-WhatIf] [-Confirm]  [<CommonParameters>]
+```
+
+### ParameterSet Channel
+
+```
+Install-PowerShell7 -Channel {Preview | LTS | Stable} [-Scope {AllUsers | CurrentUser}] [-Passthru] [-WhatIf] [-Confirm]
+[<CommonParameters>]
+```
+
+The **Version** parameter has the following validation attribute: [ValidatePattern("^7\.\d+")]
+
+The **Channel** parameter has the following validation attribute: [ValidateSet("Preview", "LTS", "Stable")]
+
+The **Scope** parameter has the following validation attribute: [ValidateSet("AllUsers", "CurrentUser")]
+
+### Data model
+
+The `Install-PowerShell7` cmdlet should support the **Passthru** parameter to provide useful
+information post installation. The data model for the output is below;
+
+```output
+PSTypename   = "PSInstallationInfo"
+[Version]Version      = $version
+[string]Path         = (Get-Command pwsh).source
+[string]Channel      = if ($Channel) { $channel} else {"Stable" }
+[string]Computername = $env:COMPUTERNAME
+```
+
+## Demo.txt
```

Comment:
What is the failure experience if running `Install-PowerShell7` and pwsh.exe is already in use so it can't be overwritten?

Reply:
Resolved by changing from cmdlet to command.